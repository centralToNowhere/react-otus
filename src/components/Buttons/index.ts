export { ButtonGameControl } from "./ButtonGameControl";
export { Button } from "./Button";
