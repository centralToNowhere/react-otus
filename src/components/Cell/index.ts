export { Cell } from "./Cell";
export type { ICell, ICellProps } from "./Cell";
