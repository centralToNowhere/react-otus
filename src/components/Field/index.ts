export { Field } from "./Field";
export type { IFiguresMap, IFieldProps } from "./Field";
