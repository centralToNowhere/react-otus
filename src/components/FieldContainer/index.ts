export {
  FieldContainer,
  createFormKey,
  getRandomCells,
  getGameCycleTimeout,
} from "./FieldContainer";
export type { FieldContainerProps } from "./FieldContainer";
