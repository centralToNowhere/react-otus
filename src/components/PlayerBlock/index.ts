export { PlayerBlock } from "./PlayerBlock";
export { PlayerBlockContainer } from "./PlayerBlockContainer";
export type { IPlayerContainerProps } from "./PlayerBlock";
