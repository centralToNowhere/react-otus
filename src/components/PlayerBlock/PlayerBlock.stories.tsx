import React from "react";
import { Story, Meta } from "@storybook/react";
import { PlayerBlock, IPlayerContainerProps } from "@/components/PlayerBlock";

export default {
  title: "Components/PlayerBlock",
  component: PlayerBlock,
  args: {
    playerName: "Player1",
    onPlayerUnregister: () => {
      //empty
    },
  },
} as Meta;

const Template: Story<IPlayerContainerProps> = (args) => {
  return <PlayerBlock {...args} />;
};

export const PlayerContainerDefault = Template.bind({});
