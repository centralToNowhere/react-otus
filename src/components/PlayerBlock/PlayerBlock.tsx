import React, { FC } from "react";
import { FormElement } from "@/components/Form";
import { css } from "@emotion/react";
import { FormField } from "@/components/Form";
import { FormButton } from "@/components/Buttons";
import { l10n } from "@/l10n/ru";
import { COLORS } from "@/styles/ui-styled";

export interface IPlayerContainerProps {
  playerName: string;
  onPlayerUnregister: () => void;
}

export const PlayerBlock: FC<IPlayerContainerProps> = (props) => {
  return (
    <FormElement
      css={css`
        border: 0;
      `}
    >
      <FormField
        css={css`
          display: inline-block;
          margin: 0;

          h1.player-name {
            font-size: 1.2rem;
            font-weight: bold;
            margin: 0;
            border-bottom: 2px solid ${COLORS.accent};
          }

          div.player-stats-container {
            margin: 20px 0;
          }

          li {
            margin: 10px 0 10px 20px;
          }
        `}
      >
        <h1 className="player-name">{props.playerName}</h1>
        <div className="player-stats-container" />
        <FormButton
          css={css`
            text-align: center;
          `}
          type="button"
          onClick={props.onPlayerUnregister}
        >
          {l10n.logoutButton}
        </FormButton>
      </FormField>
    </FormElement>
  );
};
