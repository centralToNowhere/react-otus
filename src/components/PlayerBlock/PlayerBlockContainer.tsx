import React, { FC } from "react";
import { PlayerBlock } from "@/components/PlayerBlock";
import { useDispatch, useSelector } from "react-redux";
import { logout } from "@/auth";
import { resetFieldControls } from "@/components/Fields";
import { selectPlayerName } from "@/auth";
import { resetCells } from "@/components/GameField";

export const PlayerBlockContainer: FC = () => {
  const playerName = useSelector(selectPlayerName);
  const dispatch = useDispatch();
  const onPlayerUnregister = () => {
    dispatch(logout());
    dispatch(resetCells());
    dispatch(resetFieldControls());
  };

  return (
    <PlayerBlock
      playerName={playerName}
      onPlayerUnregister={onPlayerUnregister}
    />
  );
};
