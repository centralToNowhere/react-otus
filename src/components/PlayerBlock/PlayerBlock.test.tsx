import React from "react";
import { render, screen, waitFor } from "@testing-library/react";
import { PlayerBlock } from "@/components/PlayerBlock";
import userEvent from "@testing-library/user-event/dist";

describe("PlayerContainer tests", () => {
  it("should render PlayerContainer", () => {
    jest.spyOn(global.Math, "random").mockReturnValue(0);
    const onPlayerUnregister = jest.fn();
    const { asFragment } = render(
      <PlayerBlock
        onPlayerUnregister={onPlayerUnregister}
        playerName={"Ivan"}
      />
    );
    jest.spyOn(global.Math, "random").mockRestore();

    const playerNameHeading = screen.getByRole("heading", { level: 1 });

    expect(playerNameHeading).toHaveTextContent("Ivan");
    expect(asFragment()).toMatchSnapshot();
  });

  it("should call onPlayerUnregister on player logout", async () => {
    const onPlayerUnregister = jest.fn();

    render(
      <PlayerBlock
        onPlayerUnregister={onPlayerUnregister}
        playerName={"Ivan"}
      />
    );

    const unregisterButton = screen.getByRole("button");
    userEvent.click(unregisterButton);

    await waitFor(() => {
      expect(onPlayerUnregister).toHaveBeenCalledTimes(1);
    });
  });
});
