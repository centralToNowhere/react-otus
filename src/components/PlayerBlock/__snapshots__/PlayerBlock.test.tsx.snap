// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerContainer tests should render PlayerContainer 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  border: 2px solid #6a6d75;
  border-radius: 10px;
  width: 100%;
  border: 0;
}

.emotion-0 input,
.emotion-0 label,
.emotion-0 p {
  line-height: 1;
}

.emotion-0.game-settings-form .field-error-msg {
  background: #e8d5f0;
}

.emotion-2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: 20px;
  display: inline-block;
  margin: 0;
}

@media (min-width: 576px) {
  .emotion-2 {
    -webkit-flex: 1;
    -ms-flex: 1;
    flex: 1;
  }
}

.emotion-2 input {
  margin: 5px 0 0 0;
}

.emotion-2 input,
.emotion-2 button {
  min-width: 100px;
}

.emotion-2 h1.player-name {
  font-size: 1.2rem;
  font-weight: bold;
  margin: 0;
  border-bottom: 2px solid rgb(255, 33, 92);
}

.emotion-2 div.player-stats-container {
  margin: 20px 0;
}

.emotion-2 li {
  margin: 10px 0 10px 20px;
}

.emotion-4 {
  position: relative;
  overflow: hidden;
  background: inherit;
  line-height: 1.5;
  cursor: pointer;
  border-radius: 1.5rem;
  border: 2px solid #6a6d75;
  padding: 0.4rem 1.8rem;
  text-align: center;
}

.emotion-4 span {
  display: block;
  position: absolute;
}

.emotion-4 .ripple {
  -webkit-animation: ripple 400ms linear;
  animation: ripple 400ms linear;
  background: #E59FB1;
  border-radius: 50%;
}

<form
    class="emotion-0 emotion-1"
  >
    <div
      class="emotion-2 emotion-3"
    >
      <h1
        class="player-name"
      >
        Ivan
      </h1>
      <div
        class="player-stats-container"
      />
      <button
        class="emotion-4 emotion-5"
        type="button"
      >
        <span />
        Выйти
      </button>
    </div>
  </form>
</DocumentFragment>
`;
